testapp/src/main/res/layout/settings_menu_activity.xml

```diff
@@ -58,33 +58,61 @@
+                            android:layout_height="wrap_content"
+                            android:hint="@string/hint_host_project_id"
+                            android:imeOptions="actionDone"
+                            android:inputType="text" />
+                    </com.google.android.material.textfield.TextInputLayout>
+
+                    <androidx.appcompat.widget.AppCompatImageView
+                        android:id="@+id/clearProjectId"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="end"
+                        android:layout_marginTop="@dimen/small_4"
+                        android:layout_marginEnd="@dimen/small_10"
+                        android:src="@drawable/ic_close_24dp" />
+                </FrameLayout>
 
-                <com.google.android.material.textfield.TextInputLayout
+                <FrameLayout
                     android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_margin="@dimen/small_8">
+                    android:layout_height="wrap_content">
 
-                    <androidx.appcompat.widget.AppCompatEditText
-                        android:id="@+id/editSubscriptionKey"
+                    <com.google.android.material.textfield.TextInputLayout
                         android:layout_width="match_parent"
                         android:layout_height="wrap_content"
-                        android:hint="@string/hint_subscription_key"
-                        android:imeOptions="actionDone"
-                        android:inputType="text" />
-                </com.google.android.material.textfield.TextInputLayout>
+                        android:layout_margin="@dimen/small_8">
+
+                        <androidx.appcompat.widget.AppCompatEditText
+                            android:id="@+id/editSubscriptionKey"
+                            android:layout_width="match_parent"
+                            android:layout_height="wrap_content"
+                            android:hint="@string/hint_subscription_key"
+                            android:imeOptions="actionDone"
+                            android:inputType="text" />
+                    </com.google.android.material.textfield.TextInputLayout>
+
+                    <androidx.appcompat.widget.AppCompatImageView
+                        android:id="@+id/clearSubscriptionKey"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="end"
+                        android:layout_marginTop="@dimen/small_4"
+                        android:layout_marginEnd="@dimen/small_10"
+                        android:src="@drawable/ic_close_24dp" />
+                </FrameLayout>
```

Comment:
I don't if you can use this more actual solution with `endIconMode`:

```xml
<com.google.android.material.textfield.TextInputLayout
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:hint="@string/hint_text"
    app:endIconMode="clear_text">

  <com.google.android.material.textfield.TextInputEditText
      android:layout_width="match_parent"
      android:layout_height="wrap_content"/>

</com.google.android.material.textfield.TextInputLayout>
```

Reply:
Yeah, I tried this solution before but it's breaking the UI when there is invalid input, so I am doing my own way :D 
![Screen Shot 2021-07-13 at 12 28 47](https://user-images.githubusercontent.com/66930373/125385666-e4bc0a80-e3d5-11eb-877e-d038acac2f78.png)

Reply:
you'd rather fix the invalid input icon placement in my opinion :)

Reply:
Updated this part.

Reply:
can you show a screenshot? The XML looks way better 👍

Reply:
@Climbatize I have updated the screenshot in the PR's description. 
Yeah, I like this clean solution, Thank you for your suggestions.